doc/reference/common-weaknesses/wrong-type-datum.rst

```diff
@@ -0,0 +1,19 @@
+Wrongly typed UTXO datum or redeemer
+====================================
+
+Validator scripts are usually written in a typed form, in which the types of the corresponding datum and redeemer are specified. But since both are provided using a universal “Data” type, then it is possible to supply the wrong type of data, either when creating a UTXO, or when trying to redeem one. The “typed script” API provided helps to protect against such errors, but does not exclude them altogether.
```

Comment:
I never understood this. What doesn't the typed script API prevent?

Reply:
I guess he means one can still create an arbitrary UTxO with an invalid datum?